### DISCUSSION TITLE
.env not included in the app build

### DISCUSSION THREAD

**Discussion by seve-andre** [tauri-apps/tauri#6936] [Q&A] [answered]
I'm building an app using [Diesel](https://diesel.rs/) + MySQL, and the [Getting Started](https://diesel.rs/guides/getting-started) recommends to create a `.env` file with the `DATABASE_URL`. Mine looks something like this:
```env
DATABASE_URL=mysql://mitch:password@localhost/db_name
```
Everything looks good when using `pnpm tauri dev` and `pnpm tauri build`, but after installing the `.deb` package and running the app, an error shows up:
```
thread 'tokio-runtime-worker' panicked at 'DATABASE_URL must be set: NotPresent', src/db.rs:11:49
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
So, I'm guessing the `.env` file is not included in the build, but how could I solve this?

**Comment by FabianLars** [comment, marked as answer]
You potentially could bundle it as a [resource](https://tauri.app/v1/guides/building/resources), but i _think_ this would not be next to the binary in a .deb package.

Also keep in mind that shipping .env files is often not the best idea since then all your users can just read the URL including the password.
If you don't care about that you could look at https://docs.rs/dotenvy_macro/0.15.7/dotenvy_macro/ which will include the env var from the .env file into the binary at build time (still somewhat easy to read but at least not a plain text file)

**Reply by Shubh-81** [reply]
If not shipping .env, then what is a good way to handle environment variables in built binaries?

**Reply by simantak-dabhade** [reply]
Hi yeah still trying to see, is there a better way anyone has come up with for this?

**Reply by ckhatton-pvfscaffolding** [reply]
What I found is, at build time the location of the `.env` matters - Tauri picks up the file in dev mode, but doesn't in build, which is why it might go unnoticed. So in my case, using the `dotenvy` package (with a `y`), I had to use this to point to the location of the `.env`:

```rust
dotenvy::from_filename("../.env").ok();  // Point to the root `.env`.
```

This means you will not need to ship/version-control the envs, but will have to use `dotenv` or `dotenvy` to import them as @FabianLars suggested.

**Comment by developermindsetcom** [comment]
Here's a youtube tutorial on 6 different ways how you can get environment variables set in Tauri, including `.env` files:
https://youtu.be/eFRl38t-D1U
